### DISCUSSION TITLE
Delaying the next step until event

### DISCUSSION THREAD

**Discussion by Cyleigh** [hakimio/ngx-ui-tour#183] [Q&A]
I am trying to create a tour/tutorial for a reactive form, and I'd like to be able to disable the next button until a button is clicked (the button loads more of the form).  I have the following two steps:

`  public static structuredNotesSteps: IStepOption[] = [
    {
      anchorId: 'structured.action',
      title: 'Action Note',
      allowUserInitiatedNavigation: true,
      content: '... Click the button to create an action, and then press next</b>'
    },
    {
      anchorId: 'structured.action.name',
      isAsync: true,
      asyncStepTimeout: 100,
      title: 'Action Name',
      content: 'more content'
    }
  ];`

This works fine as long as the user actually clicks the button before clicking next on the step.  However, if they simply press next and then click the button, the tour just ends (and complains about unregistered anchor).  Is there a way to easily do this or will I have to perhaps make changes to the TourService itself?

**Comment by Cyleigh** [comment]
Further investigation of the API gives me the option `nextOnAnchorClick` but this doesn't seem to work in the way that I would like.  I now have the following two steps:
`    {
      anchorId: 'structured.action',
      title: 'Action Note',
      nextOnAnchorClick: true,
      content: '.. Click the button to create an action to continue the tutorial'
    },
    {
      anchorId: 'structured.action.name',
      title: 'Action Name',
      isAsync: true,
      content: 'more content'
    },`
    
   The first step is attached to a button, which will load the next tourAnchor element into the DOM.  I get a resume event, and also the anchorRegister event, but the tour doesn't resume.

**Reply by hakimio** [reply]
Maybe you misspelled the tour anchor name in your `tourAnchor="..."` attribute?

**Reply by Cyleigh** [reply]
No.  As I said, the tour works perfectly when the user follows instructions like 'click the button, and then press next' but fails when I use `nextOnAnchorClick` (which seems to be the option I want  - pause until the button is clicked).

**Reply by hakimio** [reply]
If you want me to take a look at your issue, create a demo on StackBlitz or Github with the latest version of the library.
If library version you are using supports RxJS v6, RxJS version shouldn't be an issue.

**Reply by Cyleigh** [reply]
Thanks.  I'll see if I can recreate in a smaller project that I can share.

Edit: For what it is worth, I managed to 'fix' it by adding a `delayBeforeStepShow` to the step after the button click.  So that is fine for what I need at the moment.  Thanks

**Comment by Cyleigh** [comment]
Would having RxJs version be 6.6.0 be causing the problems do you think?

Edit: Okay.  I've updated the RxJs version and now when I click the anchor (button), the next step anchor gets highlighted but the step doesn't show.  The tour seems to get stuck and broken.

![image](https://github.com/hakimio/ngx-ui-tour/assets/11785329/c6077cdc-e224-4398-ab72-f4e0ea7d4c24)

![image](https://github.com/hakimio/ngx-ui-tour/assets/11785329/377fc8a2-9d70-46b0-b385-ee364d41cfe1)
